api/games/create.js

```diff
@@ -446,85 +447,20 @@ export const createGameFromLobby = gameLobby => {
   const { onRoundStart, onGameStart, onStageStart } = config;
   if ((onGameStart || onRoundStart || onStageStart) && firstRoundId) {
     const game = Games.findOne(gameId);
-    let gameTreatment = null,
-      gamePlayers = null,
-      gameRounds = null;
-
-    Object.defineProperties(game, {
-      treatment: {
-        get() {
-          if (!gameTreatment) {
-            gameTreatment = treatment.factorsObject();
-          }
 
-          return gameTreatment;
-        }
-      },
-      players: {
-        get() {
-          if (!gamePlayers) {
-            gamePlayers = game.getPlayers();
-            gamePlayers.forEach(player => {
-              let round = null,
-                stage = null;
-
-              Object.defineProperties(player, {
-                round: {
-                  get() {
-                    if (!round) {
-                      round = _.extend({}, nextRound);
-                    }
-
-                    return round;
-                  }
-                },
-                stage: {
-                  get() {
-                    if (!stage) {
-                      stage = _.extend({}, nextStage);
-                    }
-
-                    return stage;
-                  }
-                }
-              });
-
-              augmentPlayerStageRound(player, player.stage, player.round, game);
-            });
-          }
-
-          return gamePlayers;
-        }
-      },
-      rounds: {
-        get() {
-          if (!gameRounds) {
-            gameRounds = game.getRounds();
-            gameRounds.forEach(round => {
-              let stages = null;
-              Object.defineProperty(round, "stages", {
-                get() {
-                  if (!stages) {
-                    stages = Stages.find({ roundId: round._id }).fetch();
-                  }
-
-                  return stages;
-                }
-              });
-            });
-          }
-
-          return gameRounds;
-        }
-      }
-    });
+    augmentGameObject(game, treatment);
 
     const nextRound = game.rounds.find(r => r._id === firstRoundId);
     const nextStage = nextRound.stages.find(
       s => s._id === params.currentStageId
     );
 
     augmentGameStageRound(game, nextStage, nextRound);
+    game.players.forEach(player => {
+      player.round = _.extend({}, nextRound);
+      player.stage = _.extend({}, nextStage);
+      augmentPlayerStageRound(player, player.stage, player.round, game);
+    });
 
     if (onGameStart) {
       onGameStart(game, game.players);
```

Comment:
None of the players lazy loading is helping, given how players is called right here... (and also you added `game.players.forEach...` right above instead of keeping it inside the lazy loading bit, but that's easy to fix).
We gotta move this to the updated version of the API. Which as you can see here does not pass the `players` as the second arg, since it's already on the players on `game`.
- [ ] remove any passing of all the players to all the callbacks
- [ ] remove it from all the callbacks in the create-empirica-app template: https://github.com/empiricaly/create-empirica-app/blob/master/templates/basic/server/callbacks.js#L6
- [ ] ideally, we throw if someone tries to use that field. Or even more ideal, we proxy it transparently, with a deprecated warning (maybe using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy https://stackoverflow.com/questions/38586471/how-to-control-property-enumeration-for-in-with-proxy-objects)